src/xwayland.c

```diff
@@ -32,6 +34,17 @@ xwayland_surface_contains_window_type(
 	return false;
 }
 
+static bool
+view_contains_xwayland_window_type(struct view *view, int32_t window_type)
+{
+	assert(view);
+	if (view->type == LAB_XWAYLAND_VIEW) {
+		struct wlr_xwayland_surface *surface = xwayland_surface_from_view(view);
+		return xwayland_surface_contains_window_type(surface, window_type);
+	}
```

Comment:
No need for the `view->type` check here, if we end up in this function it must be a xwayland view as the wayland implementation (`xdg.c`) doesn't set the window pointer. If we ever end up adding something similar to the wayland variant we can always add the check back in here.

However to detect anything crazy going on I suggest adding and additional `assert(view->type == LAB_XWAYLAND_VIEW);` after the `view` assert.